unit-tests/internal/internal-tests-extrinsic.cpp

```diff
@@ -92,3 +184,254 @@ TEST_CASE("Extrinsic graph management", "[live][multicam]")
+            // 1. extrinsics table size is perserved over iterations for each stream 
+            // 2. no delay increment over iterations
+            // 3. "most" iterations have time to first frame delay below a defined threshold
+
+            std::vector<size_t> extrinsics_table_size;
+            std::map<std::string, std::vector<double>> streams_delay; // map to vector to collect all data
+            std::map<std::string, std::vector<std::map<unsigned long long, size_t >>> unique_streams_delay;
+            std::map<std::string, std::vector<unsigned long long>> frame_number;
+            std::map<std::string, size_t> new_frame;
+            std::map<std::string, size_t> extrinsic_graph_at_sensor;
+
+            auto& b = environment::get_instance().get_extrinsics_graph();
+            if (is_pipe)
+            {
+                std::cout << "==============================================" << std::endl;
+                std::cout << "Pipe Test is running .." << std::endl;
+            }
+            else {
+                std::cout << "==============================================" << std::endl;
+                std::cout << "Sensor Test is running .." << std::endl;
+            }
+
+            for (auto i = 0; i < ITERATIONS_PER_CONFIG; i++)
+            {
+                rs2::config cfg;
+                rs2::pipeline pipe;
+                size_t cfg_size = 0;
+                for (auto profile : res.second)
+                {
+                    auto fps = profile.fps;
+                    if (device_type == "D400" && profile.stream == RS2_STREAM_ACCEL) fps = 250;
+                    cfg.enable_stream(profile.stream, profile.index, profile.width, profile.height, profile.format, fps); // all streams in cfg
+                    cfg_size += 1;
+                }
+
+                for (auto it = new_frame.begin(); it != new_frame.end(); it++)
+                {
+                    it->second = 0;
+                }
+                auto t1 = std::chrono::system_clock::now().time_since_epoch();
+                bool condition = false;
+                std::mutex mutex;
+                std::mutex mutex_2;
+                auto process_frame = [&](const rs2::frame& f)
+                {
+                    std::lock_guard<std::mutex> lock(mutex_2);
+                    auto stream_type = f.get_profile().stream_name();
+                    auto frame_num = f.get_frame_number();
+                    auto time_of_arrival = f.get_frame_metadata(RS2_FRAME_METADATA_TIME_OF_ARRIVAL);
+
+                    if (std::find(frame_number[stream_type].begin(), frame_number[stream_type].end(), frame_num) == frame_number[stream_type].end())
+                    {
+                        if (!new_frame[stream_type])
+                        {
+                            frame_number[stream_type].push_back(frame_num);
+                            auto milli = std::chrono::duration_cast<std::chrono::milliseconds>(t1).count();
```

Comment:
Mentioned in previous review -
shall be moved out of scope and calculated once. 
Also rename to `start_time` for clarity